### PR TITLE
Upgrade repository

### PR DESCRIPTION
**Pull request by boryana-ilieva**
This PR adresses #3.

### CODE REVIEW

**Comment by boryana-ilieva** [comment]
Hey, greetings from New Haven!
I updated submodules and added travis integration. Last point in #3 is estimagic logging, but I would prefer to postpone this for now. It would take up a day or two and I would actually prefer to spend the upcoming days on soepy and revisit the logging next time I am estimating, when I also have a current use case. Assigning a review for this PR now therefore. Will move the logging in a new issue, to be dealt with in a future PR, if that's fine.

**Comment by boryana-ilieva** [comment]
PS. I accidentally triggered a travis build on master which is currently lacking a travis yaml. I hope the failing build status will disappear once we merge the PRs into master.

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+install:
+  - sudo apt-get update
+  - wget https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh -O miniconda.sh
+  - bash miniconda.sh -b -p $HOME/miniconda
+  - PATH="$HOME/miniconda/bin:$PATH"
+  - conda env create -f configurations/environment.yml
+  - source activate structExpectations-analysis
+
+before_script:
+  - export PYTHONPATH=$PYTHONPATH:$(pwd)
```
Comment (peisenha) [inline .travis.yml]: These two lines: This is a duplication from `.envrc`, you can simply source that file here as well. Now, I cannot simply run the same commands from the command line locally without further setup.

File: .envrc [hunk 2]
```diff
@@ -1,11 +1,10 @@
 source activate structExpectations-analysis
 
 # We need the baseline directory quite frequently
-export PROJECT_DIR=$PWD
+export PYTHONPATH=$PWD:$PYTHONPATH
```
Comment (peisenha) [inline .envrc]: Where is `PROJECT_DIR` defined now? It is used a couple of lines below. What is the motivation to change this here in the first place?

File: configurations/analysis_soepy_config.py [hunk 3]
```diff
@@ -11,5 +11,5 @@
 # We rely on relative paths throughout the package.
 PACKAGE_DIR = Path(os.path.abspath(__file__)).parents[1]
```
Comment (peisenha) [inline configurations/analysis_soepy_config.py:12]: Please rename to `PROJECT_DIR` and use our environment variable.

File: estimation/basecamp/moments_calculation.py [hunk 4]
```diff
@@ -44,7 +45,7 @@ def get_moments(data, is_store = False):
 
     # Save mean and standard deviation of wages for each period
     # to Wage Distribution section of the moments dictionary
-    for period in range(30):  ## TO DO: Remove hard coded number
+    for period in range(30):  # TO DO: Remove hard coded number
```
Comment (peisenha) [inline estimation/basecamp/moments_calculation.py]: Please endogenize or put as a module-wide variable at beginning of the module

File: python/auxiliary.py [hunk 5]
```diff
@@ -202,7 +203,7 @@ def get_moments(data):
 
     # Save mean and standard deviation of wages for each period
     # to Wage Distribution section of the moments dictionary
-    for period in range(30):  ## TO DO: Remove hard coded number
+    for period in range(30):  # TO DO: Remove hard coded number
```
Comment (peisenha) [inline python/auxiliary.py]: as above, please either endogenize or move to the top of the module.

File: setup.cfg [hunk 6]
```diff
@@ -0,0 +1,5 @@
+[flake8]
+ignore = C901, W503, E203, F841, E712
```
Comment (peisenha) [inline setup.cfg]: Too much ignoring, why not address (at least some) of them?

File: estimation/basecamp/run.py [hunk 7]
```diff
@@ -1,92 +1,35 @@
...
     get_moments=get_moments,
     log_file_name_extension=log_file_name_extension,
-    max_evals=max_evals,
 )
 
-algo_options = {"stopeval": 1e-14}
+algo_options = {"stopeval": 1e-14, "maxeval": 2}
 
 
 result = minimize(
```
Comment (peisenha) [inline estimation/basecamp/run.py:36]: Please add a simple assert that locks in the value of the criterion function going forward.

File: .travis.yml [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+script:
+    # Test code quality
+    - flake8
+    - black .
+
+    # pytest submodules
+    - py.test
+
+    # Test estimation run
+    - cd python; python create_report.py --all; cd ..
```
Comment (peisenha) [inline .travis.yml]: Please set up a link `create-report` and allow for it so be directly executed by moving it in `bin` as we do for `update-repository`.

Call `create-report` directly in `basecamp`.

**Review by peisenha — CHANGES_REQUESTED** [review]
Thanks, please mainly fix the setup of our `.envrc` file and the setup for `create-report`.

File: estimation/basecamp/moments_calculation.py (hunk 4 above)
Comment (boryana-ilieva) [reply estimation/basecamp/moments_calculation.py]: Setting to beginning of module, will fully endogenize at some future point in time

File: python/auxiliary.py (hunk 5 above)
Comment (boryana-ilieva) [reply python/auxiliary.py]: As above

File: setup.cfg (hunk 6 above)
Comment (boryana-ilieva) [reply setup.cfg]: Had a look, two down, would keep ignoring the rest.

File: configurations/analysis_soepy_config.py (hunk 3 above)
Comment (boryana-ilieva) [reply configurations/analysis_soepy_config.py:12]: Can you elaborate on this and above again: I should rename PACKAGE_DIR to PROJECT_DIR and then change what in .envrc? I am not sure what you mean by "and use our environment variable"

File: python/auxiliary.py [hunk 9]
```diff
@@ -5,6 +5,8 @@
 
 from configurations.analysis_soepy_config import RESOURCES_DIR
 
+num_periods = 30
```
Comment (peisenha) [inline python/auxiliary.py]: Please use a capitalized name for the module-wide variable. `NUM_PERIODS`. Also in the other file.

File: estimation/basecamp/moments_calculation.py [hunk 10]
```diff
@@ -45,7 +47,7 @@ def get_moments(data, is_store=False):
 
     # Save mean and standard deviation of wages for each period
     # to Wage Distribution section of the moments dictionary
-    for period in range(30):  # TO DO: Remove hard coded number
+    for period in range(num_periods):  # TO DO: Remove hard coded number
```
Comment (peisenha) [inline estimation/basecamp/moments_calculation.py]: Please use `TODO`, so PyCharm does recognize as well and provides a list.

**Comment by boryana-ilieva** [comment]
@peisenha can you leave an additional comment on the PROJECT_DIR variable? Specifically, how should it be handled in the .envrc? Thanks

**Review by peisenha — APPROVED** [review]

**Pull request merged**